src/apps/EDGE3D.cpp

```diff
@@ -0,0 +1,130 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-23, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "EDGE3D.hpp"
+
+#include "RAJA/RAJA.hpp"
+
+#include "AppsData.hpp"
+#include "common/DataUtils.hpp"
+
+#include <cmath>
+
+
+namespace rajaperf
+{
+namespace apps
+{
+
+
+EDGE3D::EDGE3D(const RunParams& params)
+  : KernelBase(rajaperf::Apps_EDGE3D, params)
+{
+  setDefaultProblemSize(100*100*100);  // See rzmax in ADomain struct
+  setDefaultReps(10);
+  constexpr int num_quadrature_points = NQ_1D*NQ_1D*NQ_1D;
+
+  Index_type rzmax = std::cbrt(getTargetProblemSize())+1;
+  m_domain = new ADomain(rzmax, /* ndims = */ 3);
+
+  m_array_length = m_domain->nnalls;
+  size_t number_of_elements = m_domain->lpz+1 - m_domain->fpz;
+
+  setActualProblemSize( number_of_elements );
+
+  setItsPerRep( number_of_elements );
+  setKernelsPerRep(1);
+
+  constexpr size_t matrix_size = NB*NB;
+  constexpr size_t basis_size = NB;
+
+  constexpr size_t reals_per_element =
+    2*NQ_1D + // quadrature size and weights
+    12*basis_size + // 3 basis, 3 tbasis, 3 dbasis, 3 tdbasis
+    matrix_size;
+
+  // touched data size, not actual number of stores and loads ?
+  setBytesPerRep( number_of_elements*reals_per_element*sizeof(Real_type) );
```

Comment:
Is this just the size of loaded data, or loaded and stored data size?

Reply:
Is this comment accurate?

Reply:
Let's go through the impl and determine that